chatbot-core/data/preprocessing/filter_processed_docs.py

```diff
@@ -0,0 +1,87 @@
+from bs4 import BeautifulSoup
+import json
+import os
+from utils import(
+    get_visible_text_length
+)
+
+SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
+INPUT_PATH = os.path.join(SCRIPT_DIR, "..", "processed", "processed_jenkins_docs.json")
+OUTPUT_PATH = os.path.join(SCRIPT_DIR, "..", "processed", "filtered_jenkins_docs.json")
+
+def link_ratio(content):
+    """
+    Computes the ratio of <a> tags to visible text chunks (roughly per 7 chars).
+
+    Parameters:
+    - content (str): Raw HTML content.
+
+    Returns:
+    - float: Ratio of links to visible text size.
+    """
+    soup = BeautifulSoup(content, "html.parser")
+    links = soup.find_all("a")
+    text = soup.get_text(separator=" ", strip=True)
+    return len(links) / max(len(text) / 7 , 1)
```

Comment:
Take care of the edge cases here `return len(links) / max(len(text) / 7 , 1)` . If text is empty, this ensures the denominator is at least 1, but it still might misrepresent the actual ratio. Do something kinda like this 

```
text_length = max(len(text.strip()), 1)
chunks = max(text_length / 7, 1)
return len(links) / chunks
```